tests/infra/base_model_tester.py

```diff
@@ -0,0 +1,117 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Callable
+
+from flax import linen, nnx
+from transformers.modeling_flax_utils import FlaxPreTrainedModel
+
+from .base_tester import BaseTester
+from .comparison import ComparisonConfig
+from .device_runner import DeviceRunner
+from .utils import Model, Workload
+
+
+class TestType(Enum):
+    INFERENCE = "inference"
+    TRAINING = "training"
+
+
+class BaseModelTester(BaseTester, ABC):
+    """
+    Abstract base class all model testers must inherit.
+
+    Derived classes must provide implementations of:
+    ```
+    _get_model() -> Model
+    _get_model_inputs() -> Sequence[Tensor]
+    _get_model_forward_pass_method_name() -> str
+    ```
+    """
+
+    def __init__(
+        self,
+        comparison_config: ComparisonConfig,
+        test_type: TestType = TestType.INFERENCE,
+    ) -> None:
+        super().__init__(comparison_config)
+        self._test_type = test_type
+
+    @staticmethod
+    @abstractmethod
+    def _get_model_as_workload() -> Workload:
```

Comment:
We should still have `get_model` and `load_model_inputs`, they should just be wrapped into workload inside this function, and this function shouldn't have to be overrided in every test, instead tests will just override and implement those former two functions.